internal/tokens/tokenmanager.go

```diff
@@ -0,0 +1,390 @@
+// specifically designed for the config environment variable so that keys can be loaded
+// from k8s or vault secrets that are mounted as files on disk
+func New(conf config.TokenConfig) (tm *TokenManager, err error) {
+	tm = &TokenManager{
+		validator: validator{
+			audience: conf.Audience,
+			issuer:   conf.Issuer,
+		},
+		conf: conf,
+		keys: make(map[ulid.ULID]*rsa.PublicKey),
+		kidEntropy: &ulid.LockedMonotonicReader{
+			MonotonicReader: ulid.Monotonic(rand.Reader, 0),
+		},
+	}
+	tm.validator.keyFunc = tm.keyFunc
+
+	for kid, path := range conf.Keys {
+		var keyID ulid.ULID
+
+		if keyID, err = ulid.Parse(kid); err != nil {
+			return nil, newParseError("path", kid, err)
+		}
+
+		// Load the keys from disk
+		var data []byte
+
+		if data, err = os.ReadFile(path); err != nil {
+			return nil, newParseError("path", kid, err)
+		}
+
+		var key *rsa.PrivateKey
+
+		if key, err = jwt.ParseRSAPrivateKeyFromPEM(data); err != nil {
+			return nil, newParseError("path", kid, err)
+		}
+
+		tm.keys[keyID] = &key.PublicKey
+
+		// Set the current key if it is the latest key
+		if tm.currentKey == nil || keyID.Time() > tm.currentKeyID.Time() {
+			tm.currentKey = key
+			tm.currentKeyID = keyID
+		}
+	}
+
+	return tm, nil
+}
+
+// NewWithKey is a constructor function that creates a new instance of the TokenManager struct
+// with a specified RSA private key. It takes in the private key as a parameter and initializes the
+// TokenManager with the provided key, along with other configuration settings from the config.TokenConfig
+// struct. It returns the created TokenManager instance or an error if there was a problem
+// initializing the TokenManager.
+func NewWithKey(key *rsa.PrivateKey, conf config.TokenConfig) (tm *TokenManager, err error) {
+	tm = &TokenManager{
+		validator: validator{
+			audience: conf.Audience,
+			issuer:   conf.Issuer,
+		},
+		conf: conf,
+		keys: make(map[ulid.ULID]*rsa.PublicKey),
+		kidEntropy: &ulid.LockedMonotonicReader{
+			MonotonicReader: ulid.Monotonic(rand.Reader, 0),
+		},
+	}
+	tm.validator.keyFunc = tm.keyFunc
+
+	var kid ulid.ULID
+
+	if kid, err = tm.genKeyID(); err != nil {
+		return nil, err
+	}
+
+	tm.keys[kid] = &key.PublicKey
+	tm.currentKey = key
+	tm.currentKeyID = kid
+
+	return tm, nil
+}
+
+// Sign an access or refresh token and return the token
+func (tm *TokenManager) Sign(token *jwt.Token) (tks string, err error) {
+	if tm.currentKey == nil || tm.currentKeyID.Compare(nilID) == 0 {
+		return "", ErrTokenManagerFailedInit
+	}
+
+	// Add the kid to the header
+	token.Header["kid"] = tm.currentKeyID.String()
+
+	// Return the signed string
+	return token.SignedString(tm.currentKey)
+}
+
+// CreateTokenPair returns signed access and refresh tokens for the specified claims in one step since usually you want both access and refresh tokens at the same time
+func (tm *TokenManager) CreateTokenPair(claims *Claims) (accessToken, refreshToken string, err error) {
```

Comment:
```suggestion
func (tm *TokenManager) CreateTokenPair(claims *Claims) (string, string, error) {
```

Either don't name them like this^ or you should just do `return` instead of
```
return accessToken, refreshToken, nil
```